### PR TITLE
refactor: access dialog

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Tidy up access dialog

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/31245893/todos/5759684550)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Shows copy journey link
- [x] Shows users with access
- [x] Shows users requesting access
- [x] Can accept users requesting access
- [x] Can promote and remove editors
- [x] Owners can't edit their own role
- [x] Editors cannot edit any roles
- [x] shows loading state when loading

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
@storyworks didn't write stories for new components since its still all covered by the original story

File: apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { GetJourneyWithUserJourneys_journey_userJourneys as UserJourney } from '../../../../__generated__/GetJourneyWithUserJourneys'
+import { UserJourneyItem } from './UserJourneyItem'
+
+interface UserJourneyListProps {
+  loading?: boolean
+  userJourneys?: UserJourney[] | null
+  disable: boolean
+}
+
+export function UserJourneyList({
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx:18]: This file probably has the more significant changes from the refactor

File: apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  loading,
+  userJourneys,
+  disable
+}: UserJourneyListProps): ReactElement {
+  const usersList: UserJourney[] = []
+  const requestsList: UserJourney[] = []
+
+  userJourneys?.forEach((userJourney) => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx:26]: Allows the component to do its own filtering for userJourneys

File: apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    if (userJourney.role === UserJourneyRole.inviteRequested) {
+      requestsList.push(userJourney)
+    } else {
+      usersList.push(userJourney)
+    }
+  })
+  return (
+    <>
+      {loading === true ? (
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx:35]: Handles the loading state

File: apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+interface Props {
+  title: string
+  users: UserJourney[]
+  disable: boolean
+}
+
+function ListSubGroup({ title, users, disable }: Props): ReactElement {
+  return (
+    <Box>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyList.tsx:85]: Box needed to group this together within the stack

File: apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyItem/UserJourneyItem.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import { RemoveUser } from './RemoveUser'
+import { ApproveUser } from './ApproveUser'
+import { PromoteUser } from './PromoteUser'
+
+interface ListItemProps {
+  userJourney: UserJourney
+  disabled: boolean
+}
+
+export function UserJourneyItem({
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/UserJourneyList/UserJourneyItem/UserJourneyItem.tsx:21]: This file is largely unchanged

File: apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx [hunk 6]
```diff
@@ -94,171 +79,27 @@ export function AccessDialog({
...
-        {!loading && (
-          <UserJourneyList
-            title="Requested Editing Rights"
-            userJourneys={data?.journey?.userJourneys?.filter(
-              ({ role }) => role === UserJourneyRole.inviteRequested
-            )}
-            disable={disable}
-          />
-        )}
+      <Stack spacing={4}>
```
Comment (csiyang) [inline apps/journeys-admin/src/components/AccessDialog/AccessDialog.tsx:82]: Refactor to use stack

**Review by storyworks — APPROVED** [review]

**Pull request merged**